packages/compiled-assets/src/index.ts

```diff
@@ -220,25 +301,73 @@ async function buildAssets(sourceDirectory: string, buildDirectory: string) {
     entryNames: '[dir]/[name]-[hash]',
     outbase: sourceDirectory,
     outdir: buildDirectory,
+    metafile: true, // Write metadata about the build
+  });
+
+  // For now, we only build ESM bundles for scripts that are split into chunks (i.e. Preact components)
+  // Using 'type=module' in the script tag for ESM means that it is loaded after all 'classic' scripts, which causes issues with bootstrap-table. See https://github.com/PrairieLearn/PrairieLearn/pull/12180.
+  const scriptBundleFiles = await globby(
+    path.join(sourceDirectory, 'scripts', 'esm-bundles', '**/*.{js,jsx,ts,tsx}'),
+  );
+  const chunkBuildResult = await esbuild.build({
```

Comment:
Nit, let's consistently call these `esmBundle*`? That matches the directory name too.

Reply:
Changed in 088f05771de6887836a96364aeaee9a82fee5021.